tutorials/CMakeLists.txt

```diff
@@ -649,7 +649,14 @@ if(ROOT_pyroot_FOUND)
                                 tutorial-pyroot-ntuple1-py)
   set(pyroot-fit1-depends tutorial-pyroot-fillrandom-py)
   set(pyroot-na49view-depends tutorial-pyroot-geometry-py)
-  set(roofit-rf104_classfactory-depends tutorial-roofit-rf104_classfactory) #Race condition
+
+  # Avoid a race condition: make sure Python tutorial is ran after C++ tutorial
+  set(roofit-rf104_classfactory-depends tutorial-roofit-rf104_classfactory)
+
+  # The rf502 rf503 python tests should not be run at the same time as the C++ tests
+  # to avoid a race condidtionrf503_workspace.root file.
+  set(roofit-rf502_wspacewrite-depends tutorial-roofit-rf503_wspaceread)
+  set(roofit-rf503_wspaceread-depends tutorial-roofit-rf502_wspacewrite-py)
```

Comment:
As Axel Said, can we avoid this cross-language dependencies? I.e. python tests generate files for other python test (if really needed) and C++ ones for other C++?
For example, what happens if `PyROOT` is disabled?

Reply:
Hi @bellenot, this is not a real cross-language dependency, but it's just that the tests for both languages can't be run at the same time because they write to the same file.

I think the simplest solution is just to name the output files for the C++ and python tutorials differently, is that okay for you?

Reply:
OK, thanks for the clarification. And yes, please. I think it's better, so they can run in parallel (and then faster on slow machines like Windows :wink:)

Reply:
Okay I'll change it! However, I think that what is called "dependency" here is not really a dependency in the true sense of the word, just a request that a test should run after another test. For example, what you called "cross-language dependecies" doesn't keep us from running only the C++ tests or only the Python tests.

I checked that by running only the C++ or only the Python tests with this PR using the following commands:
```bash
ctest -R "rf50._wsp.*py\$" .
ctest -R "rf50._wsp" -E ".*py\$" .
```